platform_login/test/test.js

```diff
@@ -15,13 +15,102 @@
+  * @param {object} aws context
+  * @param {function} callback function that returns what was passed
+  * @returns {string} callback function showing error type of badrequest has occured
+  */
+  it("should throw a BadRequest error for undefined method", function(){
+    event.method = undefined;
+    var bool = index.handler(event,context,callback).includes("Bad Request");
+    assert.isTrue(bool);
+  });
+
+  /*
+  * Given an event with non POST method, handler() shows that a Bad Request has been made
+  * @param {object} event containing a non POST method
+  * @param {object} aws context
+  * @param {function} callback function that returns what was passed
+  * @returns {string} callback function showing error type of badrequest has occured
+  */
+  it("should throw a BadRequest error for method other than POST", function(){
+    var methods = ["PUT","GET","HEAD","OPTIONS","DELETE"];
+    var badRequestBool = true;
+    //check if BadRequest occurs for each http method, if one passes without a bad request error
+    //have the above bool value be false
+    for(var i=0; i < methods.length; i++){
+      event.method = methods[i];
+      if(!index.handler(event,context,callback).includes("Bad Request")){
+        badRequestBool = false;
+      }
+    }
+    assert.isTrue(badRequestBool);
+  });
+
+  /*
+  * Given an event with no username, handler() shows that a 101 error has been made
+  * @param {object} event without username information
+  * @param {object} aws context
+  * @param {function} callback function that returns what was passed
+  * @returns {string} callback function showing username was not provided
+  */
+  it("should throw a 101 error for missing username", function(){
+    event.body.username = undefined;
+    var bool = index.handler(event,context,callback).includes("101");
+    assert.isTrue(bool);
+  });
+
+  /*
+  * Given an event with no password, handler() shows that a 102 error has been made
+  * @param {object} event without password information
+  * @param {object} aws context
+  * @param {function} callback function that returns what was passed
+  * @returns {string} callback function showing password was not provided for given user
+  */
+  it("should throw a 102 error for missing password", function(){
+    event.body.password = undefined;
+    var bool = index.handler(event,context,callback).includes("102");
+    assert.isTrue(bool);
+  });
 
-    // Add your test cases here.
-    assert(true);
-    done();
+  /*
+  * Given a valid event object, handler() should invoke cognito's authenticateUser()
+  */
+  it('should pass in user data for authentication', function() {
+    //mocking "authenticateUser" function
+    stub = sinon.stub(AWSCognito.CognitoUser.prototype, "authenticateUser", spy);
+    var returned = index.handler(event,context);
+    stub.restore();
+    assert.isTrue(spy.called);
+    //expect(() => index.handler(event,context)).to.throw();
   });

```

Comment:
One more test case to handle invalid username/password combo

Reply:
I will include this scenario; however, with the code in the index.js, once the username is found invalid an error will be thrown indicating to the user they're missing a username. Logic to validate the password wouldn't be called once the username is found invalid, so for this scenario I will be asserting that a 101 error is thrown and that a 102 was not (which may seem a bit redundant).

Reply:
user name is valid, password is valid (that is they are provided), but the combo might be wrong. [This line](https://github.com/tmobile/jazz/blob/master/platform_login/index.js#L83) is what the test case would validate.

Reply:
Ok, will include that case as well.